tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/analysis/IAnalysisModule.java

```diff
@@ -138,6 +140,41 @@ public interface IAnalysisModule extends ITmfComponent, IAnalysisRequirementProv
      */
     @Nullable Object getParameter(@NonNull String name);
 
+    /**
+     * Sets a configuration instance which provides parameters to this analysis.
+     *
+     * Use this when creating and managing analysis modules programmatically.
+     * Call the method before calling {@link #schedule()}
+     *
+     * @param configuration
+     *            A ITmfConfiguration
+     * @throws TmfConfigurationException
+     *          This exception should be thrown if configuration is invalid
+     * @since 9.5
+     */
+    default void setConfiguration(ITmfConfiguration configuration) throws TmfConfigurationException {
+        throw new TmfConfigurationException("Not implemented"); //$NON-NLS-1$
+    }
+
+    /**
+     * Gets the configuration instance used to parameterize the analysis module.
+     *
+     * @return the configuration instance or null if no configuration is set or
+     *         used
+     * @since 9.5
+     */
+    default @Nullable ITmfConfiguration getConfiguration() {
+        return null;
+    }
+
+    /**
```

Comment:
Missing a Javadoc description? Also I don't understand what this root path represents or how it is meant to be used? It's not implemented in the test.

Reply:
I forgot to remove it from another POC implementation using the analysis framework.